### DISCUSSION TITLE
Are there any requirements to be satisfied regarding the URL of a formula?

### DISCUSSION THREAD

**Discussion by andreaangiolillo** [Homebrew/discussions#3100] [Writing Formulae/Casks] [answered]
Hello,

I was wondering if there are any requirements to be satisfied regarding the URL of a formula. In particular, I have the following questions:

- Must the formula's URL be coming from the Github repository?
- Can the URL be coming from an official website such as  https://www.mongodb.com/?
- Can the archive contain a binary? Are we forced to build the formula from the source code?

Thanks!

**Comment by SMillerDev** [comment, marked as answer]
- No
- Yes
- contain, yes. Use, no.

https://docs.brew.sh/Acceptable-Formulae

**Comment by Bo98** [comment]
> Must the formula's URL be coming from the Github repository?

No

> Can the URL be coming from an official website such as https://www.mongodb.com/?

Yes. As long as:

* It's a stable, proper release.
* It's at a stable location: it won't suddenly change contents (i.e. fixed sha256) or frequently disappear.
* It is available under an open source license.

Note that for MongoDB specifically: it is not released under an [OSI-approved](https://opensource.org/licenses/alphabetical) or DFSG-compliant license so wouldn't be eligible for inclusion into homebrew-core (as is also the case with many Linux package managers).

Third party taps may have their own policies to the above.

> Can the archive contain a binary? Are we forced to build the formula from the source code?

Formulae in homebrew-core _must_ build from source. Third party taps can do as they wish.
